Polar/README.md

```diff
@@ -1,3 +1,85 @@
 # Polar Datacube Generation
 
-Information dedicated to the generation of the Polar Datacube.
\ No newline at end of file
+Information dedicated to the generation of the Polar Datacube.
+
+## Cube generation process
+
+The cube generation process is divided in four phases:
+
+> **Note**
+>
+> Before running the following scripts, you must activate the `cube-gen` conda environment by running `conda activate cube-gen`.
+
+### 1. Downloading required raw datasets
+
+> **Note**
+>
+> When downloading the datasets, a path `data/ocean/source/` is created in the user home path. All datasets are downloaded into this folder for further preprocessing.
+
+Datasets were downloaded via the source information provided in `cube.geojson`. Datasets were downloaded in different formats: `.nc`, `.tif`, `.shp`. Some datasets were compressed and this step includes the extraction of the data.
+
+To download the datasets it is required to run the following file:
+
+```
+# Download all files
+python inputs-collect/download-sources.py
+```
+
+### 2. Preprocessing datasets
+
+The raw data is preprocessed in order to create a single input `.zarr` cube per dataset. The preprocessing steps might involve time resampling and/or spatial resampling according to the dataset. The preprocessing code for each dataset is found at the `inputs-preprocess` folder. Note that some datasets have multiple parts according to the preprocessing steps that were applied:
+
+```
+# Curie Depth Estimates
+python inputs-preprocess/curie-depth.py # Convert files to zarr
+python inputs-preprocess/curie-depth-resample.py # Spatial resampling
+
+# Geothermal Heat Flow
+python inputs-preprocess/geothermal-heat-flow.py # Convert files to zarr
+python inputs-preprocess/geothermal-heat-flow-resample.py # Spatial resampling
+
+# Geothermal Heat Flow Uncertainty
+python inputs-preprocess/geothermal-heat-flow-uncertainty.py # Convert files to zarr
+python inputs-preprocess/geothermal-heat-flow-uncertainty-resample.py # Spatial resampling
+
+# Magnetic Anomaly
+python inputs-preprocess/magnetic-anomaly.py # Convert files to zarr
+python inputs-preprocess/magnetic-anomaly-resample.py # Spatial resampling
+
+# Subglacial Lakes
+python inputs-preprocess/subglacial-lakes-2013-2017.py # Convert files to zarr
```

Comment:
(cube-gen env is activated)

The command 

`python inputs-preprocess/subglacial-lakes-2013-2017.py`

fails with: 

```
Traceback (most recent call last):
  File "/home/alicebalfanz/cube-gen/Polar/inputs-preprocess/subglacial-lakes-2013-2017.py", line 45, in <module>
    da.to_zarr(f"{pathOut}/thwaites_sublacial_lakes_2013_2017.zarr")
  File "/home/conda/deepesdl/envs/cube-gen/lib/python3.9/site-packages/xarray/core/dataset.py", line 2068, in to_zarr
    return to_zarr(  # type: ignore
  File "/home/conda/deepesdl/envs/cube-gen/lib/python3.9/site-packages/xarray/backends/api.py", line 1653, in to_zarr
    dump_to_store(dataset, zstore, writer, encoding=encoding)
  File "/home/conda/deepesdl/envs/cube-gen/lib/python3.9/site-packages/xarray/backends/api.py", line 1273, in dump_to_store
    store.store(variables, attrs, check_encoding, writer, unlimited_dims=unlimited_dims)
  File "/home/conda/deepesdl/envs/cube-gen/lib/python3.9/site-packages/xarray/backends/zarr.py", line 540, in store
    variables_encoded, attributes = self.encode(
  File "/home/conda/deepesdl/envs/cube-gen/lib/python3.9/site-packages/xarray/backends/common.py", line 202, in encode
    variables = {k: self.encode_variable(v) for k, v in variables.items()}
  File "/home/conda/deepesdl/envs/cube-gen/lib/python3.9/site-packages/xarray/backends/common.py", line 202, in <dictcomp>
    variables = {k: self.encode_variable(v) for k, v in variables.items()}
  File "/home/conda/deepesdl/envs/cube-gen/lib/python3.9/site-packages/xarray/backends/zarr.py", line 499, in encode_variable
    variable = encode_zarr_variable(variable)
  File "/home/conda/deepesdl/envs/cube-gen/lib/python3.9/site-packages/xarray/backends/zarr.py", line 285, in encode_zarr_variable
    var = conventions.encode_cf_variable(var, name=name)
  File "/home/conda/deepesdl/envs/cube-gen/lib/python3.9/site-packages/xarray/conventions.py", line 273, in encode_cf_variable
    var = coder.encode(var, name=name)
  File "/home/conda/deepesdl/envs/cube-gen/lib/python3.9/site-packages/xarray/coding/times.py", line 673, in encode
    (data, units, calendar) = encode_cf_datetime(
  File "/home/conda/deepesdl/envs/cube-gen/lib/python3.9/site-packages/xarray/coding/times.py", line 609, in encode_cf_datetime
    units = infer_datetime_units(dates)
  File "/home/conda/deepesdl/envs/cube-gen/lib/python3.9/site-packages/xarray/coding/times.py", line 391, in infer_datetime_units
    reference_date = format_cftime_datetime(reference_date)
  File "/home/conda/deepesdl/envs/cube-gen/lib/python3.9/site-packages/xarray/coding/times.py", line 402, in format_cftime_datetime
    date.year,
AttributeError: 'numpy.datetime64' object has no attribute 'year'
```

Reply:
Corrected!